scenes/hud.gd

```diff
@@ -0,0 +1,21 @@
+extends CanvasLayer
+@onready var pause_menu = $PauseMenu
+var paused = true
+
+func _ready():
+	paused = true
+	pauseMenu()
+
+func _process(delta):
+	if Input.is_action_just_pressed("pause"):
+		pauseMenu()
+
+func pauseMenu():
+	paused = !paused
+	if paused:
+		pause_menu.show()
+		Engine.time_scale = 0
```

Comment:
joshua says this is wrong. (also freezes menu) freeze should applied to the tree